lib/broadway_dashboard.ex

```diff
@@ -78,6 +98,18 @@ defmodule BroadwayDashboard do
         to = live_dashboard_path(socket, socket.assigns.page, nav: first_pipeline)
         {:ok, push_redirect(socket, to: to)}
 
+      pipelines == [] ->
+        error =
+          case auto_discover_result do
+            :ok ->
+              :no_pipelines_available
+
+            {:error, error} ->
+              error
+          end
+
+        {:ok, assign(socket, pipeline: nil, error: error)}
+
       true ->
         {:ok, assign(socket, pipeline: nil, error: :pipeline_not_found)}
```

Comment:
I think we can rewrite this cond. For example, this last clauses says pipeline_not_found, but if the user passes a nav and there is no pipeline, the first clause already catches that? So pipelines not found are already handled uptop?

What about this:

```elixir
    cond do
      # We couldn't discover or pipeline is empty
      auto_discover_result != :ok ->
        {:error, error} = auto_discover_result
        {:ok, assign(socket, pipeline: nil, error: error)}

      # We found the user's pipeline
      pipeline && connected?(socket) ->
        node = socket.assigns.page.node
        with :ok <- check_broadway_version(node),
             {:ok, initial_payload} <- Metrics.listen(node, self(), pipeline) do
          stats = %{
            successful: initial_payload.successful,
            failed: initial_payload.failed,
            throughput_successful: 0,
            throughput_failed: 0
          }
          layers = PipelineGraph.build_layers(initial_payload.topology_workload)
          {:ok, assign(socket, pipeline: pipeline, stats: stats, layers: layers)}
        else
          {:error, error} ->
            {:ok, assign(socket, pipeline: nil, error: error)}
        end

      # The user chose a pipeline but we can't find it or none was chosen
      true ->
        to = live_dashboard_path(socket, socket.assigns.page, nav: hd(pipelines))
        {:ok, push_redirect(socket, to: to)}
    end
```

The first clause requires auto discover to returns error if pipeline is empty.

Reply:
@josevalim thank you for the suggestion! I did slightly different: I'm checking the result of pipeline discovery before the `cond`. It's easier to reason the flow.

PS: it was catching that scenarios. I made a small change to simplify as you suggested.